src/meta/src/stream/stream_graph.rs

```diff
@@ -778,6 +819,33 @@ impl ActorGraphBuilder {
 
         let parallel_degree = if current_fragment.is_singleton {
             1
+        } else if fragment_graph
```

Comment:
https://github.com/risingwavelabs/risingwave/pull/7490 is almost ready. 🤤 Hope that will significantly reduce the complexity here.

Reply:
Look forward to it!🤤